lib/cyclonedx/base.rb

```diff
@@ -24,20 +25,32 @@ def build_metadata
     # Returns the 'components' object for a supported/unsupported scanner's report
     def build_components_object
       components = []
-      @scan_report.info[:dependencies].each do |dependency|
-        component = {
-          "bom-ref": "",
-          "type": DEFAULT_COMPONENT_TYPE,
-          "group": "",
-          "name": dependency[:name],
-          "version": "",
-          "purl": ""
-        }
-
-        # TODO: Add specific component parsing for individual scanners
-        components << component
+      info = @scan_report.to_h.fetch(:info)
+      info[:dependencies].each do |dependency|
+        components << parse_dependency(dependency)
       end
       components
     end
+
+    def parse_dependency(dependency)
+      {
+        "bom-ref": package_url(dependency),
+        "type": DEFAULT_DEP_COMPONENT_TYPE,
+        "group": "", # TODO: add group or domain name of the publisher
+        "name": dependency[:name],
+        "version": version_string(dependency),
+        "purl": package_url(dependency),
+        "properties": [
+          {
+            "key": "source",
+            "value": dependency[:source]
+          },
+          {
+            "key": "dependency_file",
+            "value": dependency[:dependency_file]
```

Comment:
There is an option to have a `build-meta` external reference if you wanted to provide a way to retrieve the gemfile that was used.

Reply:
Thanks for the insight! This property is currently being used to specify which file the dependencies were parsed from (i.e Gemfile vs Gemfile.lock). Will keep the build-meta external reference in mind if we want to provide a url to the dependency file used.